tests/simple/simple.go

```diff
@@ -134,8 +134,18 @@ func MatchValue(tag string, expected *exprpb.Value, actual *exprpb.Value) error
 		}
 	default:
 		// By default, just compare the protos.
-		if !proto.Equal(expected, actual) {
-			return fmt.Errorf("%s: Eval got [%v], want [%v]", tag, actual, expected)
+		// Compare the canonical string marshaling which is closer
+		// to protobuf equality semantics than proto.Equal:
+		// - properly compares Any messages, which might be
+		//   equivalent even with different byte encodings;
+		// - surfaces sign differences for floating-point zero.
+		// Text marshaling isn't documented as deterministic,
+		// but it appears to be so in practice.
```

Comment:
Interesting. I hope this property holds.

Reply:
Text serialization is deterministic (i.e. produces the same bytes for the same message within the same program), so this hack works. It is definitely not stable though (i.e., produces the same bytes for the same message for all time).

Longer term, you'll want to use the [v2 protocmp library](https://pkg.go.dev/google.golang.org/protobuf/testing/protocmp?tab=doc), which is designed for comparing protobuf messages in tests:
```go
cmp.Equal(want, got, protocmp.Transform())
```